templates/cpp-template-default/proj.android-studio/app/build.gradle

```diff
@@ -18,10 +18,10 @@ android {
                 if (!project.hasProperty("PROP_NDK_MODE") || PROP_NDK_MODE.compareTo('none') != 0) {
                     // skip the NDK Build step if PROP_NDK_MODE is none
                     targets 'MyGame'
-                    arguments 'NDK_TOOLCHAIN_VERSION=4.9'
-                    arguments 'APP_PLATFORM=android-'+PROP_APP_PLATFORM
+                    arguments 'NDK_TOOLCHAIN_VERSION=clang'
+                    arguments 'APP_PLATFORM=android-' + PROP_MIN_SDK_VERSION
```

Comment:
Should `APP_PLATFORM` for building c++ code also use `PROP_SDK_VERSION`?
If no, why?

Reply:
Since there are some parts of the NDK that are only available to newer version of Android (Example: https://developer.android.com/ndk/guides/neuralnetworks/index.html. This is only available for devices running Oreo) You want to compile the `APP_PLATFORM` to the lower bound of your application. In this case it will be whatever the `minSdkVersion` is.

While I doubt people will be using the NDK directly while making a game, This is makes sure that the default project don't

Reply:
Yep. 
@minggo , so we need to add this changes in CHANGELOG while releasing next version `v3.17`.
`PROP_APP_PLATFORM` was gone.
`PROP_COMPILE_SDK_VERSION` and `PROP_TARGET_SDK_VERSION` were also gone and replaced with `PROP_SDK_VERSION`.

Reply:
I did not test with the latest NDK. But I believe that APP_PLATFORM should be = PROP_TARGET_SDK_VERSION. It's just logical.

Theoretically PROP_MIN_SDK_VERSION will work. But then we do not get the benefits of a newer NDK.

@minggo @dumganhar

Reply:
@Priority10 Any links? I think that safer is APP_PLATFORM==PROP_MIN_SDK_VERSION ?

Reply:
Maybe. In any case, PROP_TARGET_SDK_VERSION=XX implies that you have tested on devices with PROP_MIN_SDK_VERSION to XX and everything works.
If you do not want to test, then yes it is safer.

I do not want to look for links right now, just read the official documentation.

Reply:
@mozartalouis, could we keep `PROP_COMPILE_SDK_VERSION`, `PROP_TARGET_SDK_VERSION` and make them the same value in the test projects and template projects?
It's will be more customizable if developer has special needs.

Reply:
I read the comments bellow.
Yep, it's difficult to make a choice.
Personally, I like the idea of keeping the old PROP and set them the same value of the latest SDK version better.